### PR TITLE
Adding Strongholds

### LINKED ISSUE #14: Add strongholds (Fixed Map)
Each stronghold will have:
  * soldier count
  * owner

### PR DESCRIPTION
**Pull request by QUzair**
Closes #14 

-  Created Simple Map
- Added Prefab to scale strongholds, magnitude and color (For testing purposes)
- Added Player Class to instantiate color and name
- Added Rendering Component to change color

### CODE REVIEW

File: Assets/Stronghold_State.cs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        }
+        else
+        {
+            Debug.Log("Player doesnt exist");
+        }
+
+    }
+	
+	// Update is called once per frame
+	void Update () {
```
Comment (salahusama) [inline Assets/Stronghold_State.cs]: Since this isn't being used, I think we can remove it for now.

File: Assets/Stronghold_State.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        rend = GetComponent<Renderer>();
+    
+        //Scale StrongHold to Required User Value
+        transform.localScale =new Vector3(scale,scale,scale);
+
+        //Instantiate Color and Magnitude
+        if (player != null)
+        {
+            Debug.Log("Changing Color");
+            rend.material.color = player.color;
```
Comment (salahusama) [inline Assets/Stronghold_State.cs]: Nice 💯

File: Assets/Stronghold_State.cs [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    // Use this for initialization
+    void Start() {
+
+        //Get Rendering Component
+        rend = GetComponent<Renderer>();
+    
+        //Scale StrongHold to Required User Value
+        transform.localScale =new Vector3(scale,scale,scale);
+
+        //Instantiate Color and Magnitude
```
Comment (salahusama) [inline Assets/Stronghold_State.cs]: Only color was instantiated 👀

File: Assets/Stronghold_State.cs [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    public Renderer rend;
+
+    // Use this for initialization
+    void Start() {
+
+        //Get Rendering Component
+        rend = GetComponent<Renderer>();
+    
+        //Scale StrongHold to Required User Value
+        transform.localScale =new Vector3(scale,scale,scale);
```
Comment (salahusama) [inline Assets/Stronghold_State.cs]: What's this for?

File: Assets/Stronghold_State.cs [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Stronghold_State : MonoBehaviour {
+
+    //Player color and name Properties
+    public Player player;
+
+    //Keep for testing Purposes of Mechanics *NOT NECESSARY*
+    public string color;
```
Comment (salahusama) [inline Assets/Stronghold_State.cs]: I don't see this being used anywhere. Is it needed?

File: Assets/Stronghold_State.cs [hunk 6]
```diff
@@ -0,0 +1,46 @@
+﻿using System.Collections;
+using System.Collections.Generic;
```
Comment (salahusama) [inline Assets/Stronghold_State.cs]: Nitpick: It's not important, but if we're not using these, I think we can remove them.

**Comment by salahusama** [comment]
> Also @salahusama it does say null, but with a violation error. Hence I created the third Empty Stronghold object which makes it work, but can't be dragged into prefab, only individual objects.

I pulled this locally and I don't see an error 🤔 I also don't see an empty Stronghold object.

File: Assets/Stronghold_State.cs (hunk 1 above)
Comment (QUzair) [reply Assets/Stronghold_State.cs]: Yeah we can replace it with onCollisionEnter() when we are implementing collisions.

**Pull request merged**